examples/01-module-declarations-reified-as-module-objects.md

```diff
@@ -0,0 +1,44 @@
+# Module Declarations reified as Module Objects
+
+$$ {\text{module declarations} \over \text{module expressions}} = {\text{function declarations} \over \text{function expressions}} $$
+
+Except for the "they are static declaration parts", module declarations behave exactly like module expressions assigned to a `const` variable.
```

Comment:
so, like const/let - the declaration hoists to the top of the block, and the initialization does not hoist, creating a TDZ between the start of the block and the module declaration?

Reply:
I pushed [`eb53890` (#21)](https://github.com/tc39/proposal-module-fragments/pull/21/commits/eb5389017e16c02930870e83b8391d6407ee258b) to clarify it: yes you can use them before their definition, but it's more like `function` than like `var` (because their value is already present).

Reply:
and it has function scope or block scope?

Reply:
why would module declarations - or expressions - only be able to appear at the top level? They can be used in expression position with dynamic import.

Reply:
Module expressions can appear anywhere. Module declarations can only appear at the top-level because the only benefit that module declarations give compared to module expressions is that you can statically import them, and static imports can only appear at the top level. Also, since module declarations don't capture variables you can easily manually move them to the top level.

I don't have a good reason to disallow them in any position other than "it's unnecessary, and it makes the static analysis of visible module declarations harder": I put the simpler version in this first version of the spec text to be able to focus on the major semantics, but please consider this as still up for debate. If there is an use case for module declarations in arbitrary position then I'm happy to support them.